### PR TITLE
Eliminate `errs.Matches()`.

### PR DESCRIPTION
**Pull request by mitchell-as**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-2775" title="DX-2775" target="_blank">TaskDX-2775</a>  Get rid of `errs.Matches()`
  </td></table>
  <br />
 

Use the more-explicit and less-error-prone `errors.As()` instead.

### CODE REVIEW

File: internal/runners/commit/commit.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func rationalizeError(err *error) {
 			"Your buildscript contains no new changes. No commit necessary.",
 		), errs.SetInput())
 
-	case errs.Matches(*err, buildscript_runbit.ErrBuildscriptNotExist):
+	case errors.Is(*err, buildscript_runbit.ErrBuildscriptNotExist):
```
Comment (mitchell-as) [inline internal/runners/commit/commit.go:53]: The ticket served its purpose, because we caught this one!

**Comment by mitchell-as** [comment]
The Windows test failure is a sporadic one and unrelated to this PR.

**Review by MDrakos — DISMISSED** [review]

**Comment by mitchell-as** [comment]
@MDrakos please re-approve. I had to fix a merge conflict with an incoming comma-splice fix. There are no functional changes from what you already reviewed. Thanks!

**Comment by mitchell-as** [comment]
Oh never mind, it looks like I can put this one in myself.

**Pull request merged**